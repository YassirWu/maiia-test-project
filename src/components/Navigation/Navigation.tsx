import React from 'react';
import Navbar from 'react-bootstrap/Navbar';

import { makeStyles } from '@material-ui/styles';

const useStyle = makeStyles({
  root: {
    marginBottom: '20px',
  },
})

type NotificationProps = {
  totalShoppingCart: number,
}

const Navigation: React.FunctionComponent<NotificationProps> = ({ totalShoppingCart }) => {
  const classes = useStyle();

  return (
    <Navbar className={classes.root}>
      <Navbar.Brand>Product Website</Navbar.Brand>
      <Navbar.Text>Panier {totalShoppingCart}</Navbar.Text>
    </Navbar>
  );
};

export default Navigation;
