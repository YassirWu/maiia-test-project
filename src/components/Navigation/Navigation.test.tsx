import React from 'react';
import { shallow } from 'enzyme';
import Navigation from './Navigation';

describe('<Navigation />', () => {
  it('should render Navigation component', () => {
    const wrapper = shallow(
      <Navigation
        totalShoppingCart={20}
        onClickShoppingCart={jest.fn()}
        searchByName={jest.fn()}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
})


