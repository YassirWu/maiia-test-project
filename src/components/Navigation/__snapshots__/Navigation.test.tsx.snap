// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navigation /> should render Navigation component 1`] = `ShallowWrapper {}`;
